src/Field/Field.hh

```diff
@@ -236,6 +242,36 @@ public:
   // Functions to help with storing the field in a Sidre datastore.
   axom::sidre::DataTypeId getAxomTypeID() const;
 
+  using ViewType = FieldView<Dimension, DataType>;
+
+
+
+  ViewType toView()
+  {
+    return this->toView(
+      [](const chai::PointerRecord *,
+        chai::Action action,
+        chai::ExecutionSpace) {
+
+        if (action == chai::ACTION_MOVE)
+          std::cout << "Chai ManagedArray Moved.\n";
+        }
+
+    );
+  }
+
+  template<typename F>
+  ViewType toView(F callback)
+  {
+    if (!mManagedData.getActivePointer()) {
+      mManagedData = chai::makeManagedArray(
+          mDataArray.data(), mDataArray.size(), chai::CPU, false);
+
+      mManagedData.setUserCallback(callback);
+    }
+
+    return ViewType(mManagedData);
+  }
```

Comment:
TODO: Need to add conditions to check if the host side vector allocation has changed before returning data.